my_pyscf/mcpdft/laspdft.py

```diff
@@ -32,31 +32,45 @@ def get_h2eff(self, mo_coeff=None):
             eri = ao2mo.full(self.mol, mo_coeff, verbose=self.verbose,
                                 max_memory=self.max_memory)
         return eri
-
-def get_mcpdft_child_class(mc, ot, **kwargs):
+        
+def get_mcpdft_child_class(mc, ot, DoLASSI=False,  **kwargs):
     mc_doc = (mc.__class__.__doc__ or 'No docstring for MC-SCF parent method')
    
     class PDFT(_LASPDFT, mc.__class__):
         __doc__= mc_doc + '\n\n' + _LASPDFT.__doc__
         _mc_class = mc.__class__
+        setattr(_mc_class, 'DoLASSI', None)
 
         def get_h2eff(self, mo_coeff=None):
             if self._in_mcscf_env: return mc.__class__.get_h2eff(self, mo_coeff=mo_coeff)
             else: return _LASPDFT.get_h2eff(self, mo_coeff=mo_coeff)
-
-        make_one_casdm1s=mc.__class__.make_casdm1s
-        make_one_casdm2=mc.__class__.make_casdm2
         
+        if DoLASSI:  _mc_class.DoLASSI = True
+        else: _mc_class.DoLASSI = False
+
+        if _mc_class.DoLASSI:
+            # This code doesn't seem efficent, have to calculate the casdm1 and casdm2 in different functions.
```

Comment:
Yes it's inefficient, but the way to resolve the inefficiency is to refactor the underlying LASSI function `root_make_rdm12s`, not to cache stuff. Cacheing intermediates is dangerous.

Reply:
This is just a response to your comment, I'm not asking you to do this before I accept the PR